format/macho/macho.go

```diff
@@ -0,0 +1,851 @@
+				d.FieldU64("reserved4")
+				d.FieldU64("reserved5")
+				d.FieldU64("reserved6")
+			}
+		case LC_SUB_UMBRELLA, LC_SUB_LIBRARY, LC_SUB_CLIENT, LC_SUB_FRAMEWORK:
+			offset := d.FieldU32("offset")
+			d.FieldUTF8NullFixedLen("name", int(cmdsize)-int(offset))
+		case LC_SYMTAB:
+			d.FieldU32("symoff")
+			d.FieldU32("nsyms")
+			d.FieldU32("stroff")
+			d.FieldU32("strsize")
+		case LC_DYSYMTAB:
+			d.FieldU32("ilocalsym")
+			d.FieldU32("nlocalsym")
+			d.FieldU32("iextdefsym")
+			d.FieldU32("nextdefsym")
+			d.FieldU32("iundefsym")
+			d.FieldU32("nundefsym")
+			d.FieldU32("tocoff")
+			d.FieldU32("ntoc")
+			d.FieldU32("modtaboff")
+			d.FieldU32("nmodtab")
+			d.FieldU32("extrefsymoff")
+			d.FieldU32("nextrefsyms")
+			d.FieldU32("indirectsymoff")
+			d.FieldU32("nindirectsyms")
+
+			d.FieldU32("extreloff")
+			d.FieldU32("nextrel")
+			d.FieldU32("locreloff")
+			d.FieldU32("nlocrel")
+		case LC_BUILD_VERSION:
+			d.FieldU32("platform")
+			d.FieldU32("minos")
+			d.FieldU32("sdk")
+			ntools := d.FieldU32("ntools")
+			var ntoolsIdx uint64
+			d.FieldStructArrayLoop("tools", "tool", func() bool {
+				return ntoolsIdx < ntools
+			}, func(d *decode.D) {
+				d.FieldU32("tool")
+				d.FieldU32("version")
+				ntoolsIdx++
+			})
+		case LC_CODE_SIGNATURE, LC_SEGMENT_SPLIT_INFO, LC_FUNCTION_STARTS, LC_DATA_IN_CODE, LC_DYLIB_CODE_SIGN_DRS, LC_LINKER_OPTIMIZATION_HINT:
+			d.FieldStruct("linkedit_data", func(d *decode.D) {
+				d.FieldU32("off")
+				d.FieldU32("size")
+			})
+		case LC_VERSION_MIN_IPHONEOS, LC_VERSION_MIN_MACOSX, LC_VERSION_MIN_TVOS, LC_VERSION_MIN_WATCHOS:
+			d.FieldU32("version")
+			d.FieldU32("sdk")
+		case LC_DYLD_INFO, LC_DYLD_INFO_ONLY:
+			d.FieldStruct("dyld_info", func(d *decode.D) {
+				d.FieldU32("rebase_off")
+				d.FieldU32("rebase_size")
+				d.FieldU32("bind_off")
+				d.FieldU32("bind_size")
+				d.FieldU32("weak_bind_off")
+				d.FieldU32("weak_bind_size")
+				d.FieldU32("lazy_bind_off")
+				d.FieldU32("lazy_bind_size")
+				d.FieldU32("export_off")
+				d.FieldU32("export_size")
+			})
+		case LC_MAIN:
+			d.FieldStruct("entrypoint", func(d *decode.D) {
+				d.FieldU64("entryoff")
+				d.FieldU64("stacksize")
+			})
+		case LC_SOURCE_VERSION:
+			d.FieldStruct("source_version_tag", func(d *decode.D) {
+				d.FieldU64("tag")
+			})
+		case LC_LINKER_OPTION:
+			d.FieldStruct("linker_option", func(d *decode.D) {
+				count := d.FieldU32("count")
+				d.FieldUTF8NullFixedLen("option", int(count))
+			})
+		case LC_ENCRYPTION_INFO, LC_ENCRYPTION_INFO_64:
+			d.FieldStruct("encryption_info", func(d *decode.D) {
+				d.FieldU32("offset")
+				d.FieldU32("size")
+				d.FieldU32("id")
+			})
+		case LC_IDFVMLIB, LC_LOADFVMLIB:
+			d.FieldStruct("fvmlib", func(d *decode.D) {
+				offset := d.FieldU32("offset")
+				d.FieldU32("minor_version")
+				d.FieldU32("header_addr", scalar.Hex)
+				d.FieldUTF8NullFixedLen("name", int(cmdsize)-int(offset))
+			})
+		default:
+			if _, ok := loadCommands[cmd]; !ok {
```

Comment:
Is there any command not handled in a cases above that are in loadCommands? if so i think they will be skipped and get no unknown field? btw i've used "unknown" fields for gaps that fq finds automatically so maybe good to use some other word like "data"?
If something is truly unknown the idea is for a decoder not even add any fields for it and let fq add gap fields instead

Reply:
Thinking it might be better to add this as "data" etc instead of seeking paste by it. Sorry maybe should have explained more in details what the idea of auto "unknown" fields are usually used for. For example in this case we know the command number and the size so we could add "data" field saying these bits belong to the command at least. But yes it's a bit of balance sometimes what to skip , what to add as something close to "unknown".

Reply:
What do you think about this? maybe most of the unknown i saw as actually just commands that are handled but the offset/size fields are not used at the moment? for ex LC_CODE_SIGNATURE and some others (hard to do generic?) there is offset and size that could be added a as raw "data". Maybe for later?